libsolidity/analysis/FunctionCallGraph.h

```diff
@@ -0,0 +1,73 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libsolidity/ast/ASTVisitor.h>
+#include <memory>
+
+namespace solidity::frontend
+{
+
+// Traverses the constructors and state variable assignments to builds a
+// function call graph
+class FunctionCallGraphBuilder : private ASTConstVisitor
```

Comment:
```suggestion
class FunctionCallGraphBuilder: private ASTConstVisitor
```
not sure but I think this is the style?